From: Folia Compatibility Patch
Subject: [PATCH 0013] DisableKnockbackEffect: 

AE PATCH REASON: 
AE PATCH FIX: 
AE PATCH RISK: 

---
diff --git a/net/advancedplugins/ae/impl/effects/effects/effects/internal/DisableKnockbackEffect.java b/net/advancedplugins/ae/impl/effects/effects/effects/internal/DisableKnockbackEffect.java
--- a/net/advancedplugins/ae/impl/effects/effects/effects/internal/DisableKnockbackEffect.java	2026-07-30 10:27:31.664350087 +0000
+++ b/net/advancedplugins/ae/impl/effects/effects/effects/internal/DisableKnockbackEffect.java	2026-07-30 12:22:21.509498105 +0000
@@ -4,8 +4,8 @@
 import java.util.UUID;
 import net.advancedplugins.ae.impl.effects.effects.actions.execution.ExecutionTask;
 import net.advancedplugins.ae.impl.effects.effects.effects.AdvancedEffect;
+import io.papermc.paper.threadedregions.scheduler.ScheduledTask;
 import net.advancedplugins.ae.impl.utils.ASManager;
-import net.advancedplugins.ae.impl.utils.SchedulerUtils;
 import org.bukkit.entity.LivingEntity;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.EventPriority;
@@ -39,7 +39,7 @@
             if (var3 - System.currentTimeMillis() <= 0L) {
                this.entityMap.remove(var2.getUniqueId());
             } else {
-               SchedulerUtils.runTaskLater(() -> var2.setVelocity(new Vector(0.0F, 0.0F, 0.0F)));
+                var2.getScheduler().runDelayed(ASManager.getInstance(), task -> var2.setVelocity(new Vector(0, 0, 0)), null, 1L);
             }
          }
       }
